core-bundle/src/Controller/SitemapController.php

```diff
@@ -142,7 +142,7 @@ private function getPageAndArticleUrls(int $parentPageId): array
 
             $isPublished = $pageModel->published && (!$pageModel->start || $pageModel->start <= time()) && (!$pageModel->stop || $pageModel->stop > time());
 
-            if ($isPublished && !$pageModel->requireItem && $this->pageRegistry->supportsContentComposition($pageModel)) {
+            if ($isPublished && !$pageModel->requireItem && $this->pageRegistry->supportsContentComposition($pageModel) && 'noindex,nofollow' !== $pageModel->robots) {
```

Comment:
```suggestion
            if ($isPublished && !$pageModel->requireItem && 'noindex,nofollow' !== $pageModel->robots && $this->pageRegistry->supportsContentComposition($pageModel)) {
```
Maybe more performant this way?